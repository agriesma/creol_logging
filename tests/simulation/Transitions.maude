***
*** Datatypes.maude -- Test suite for the data types module.
***
*** Copyright (c) 2007
***
*** The purpose of this file is to create the files `prelude.creol'
*** and `modelcheck.creol'.  These files have to be processed with
*** m4, with either one of `CREOL' or `MODELCHECK' defined.
***
*** This program is free software; you can redistribute it and/or
*** modify it under the terms of the GNU General Public License as
*** published by the Free Software Foundation; either version 2 of the
*** License, or (at your option) any later version.
***
*** This program is distributed in the hope that it will be useful, but
*** WITHOUT ANY WARRANTY; without even the implied warranty of
*** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
*** General Public License for more details.
***
*** You should have received a copy of the GNU General Public License
*** along with this program; if not, write to the Free Software
*** Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
*** 02111-1307, USA.
***

set show timing off .
set show advise off .
load creol-logginginterpreter


rew getThis(("s" |-> int(2), "this" |-> ob("myglobal"))) .
rew getLabel(("s" |-> int(2), "this" |-> ob("myglobal"))) .

rew replacementMap(("s" |-> int(2), "this" |-> ob("myglobal")), ("f" |-> int(3), ".label" |-> label(ob("mymethod"), 1) ) ) .
rew replacementMap(("s" |-> int(2)), ("f" |-> int(3), ".label" |-> label(ob("mymethod"), 1) ) ) .
rew replacementMap(("s" |-> int(2), "this" |-> ob("myglobal")), ("f" |-> int(3) ) ) .
rew replacementMap(("s" |-> int(2), "this" |-> ob("myglobal")), ("s" |-> int(2), "f" |-> int(3), ".label" |-> label(ob("mymethod"), 1) ) ) .

rew replace ("&&"("<"("m" :: "mmax") :: "<"("-"("mfree" :: "t") :: "/"("m" :: "mrate"))), "m", "nte") .

rew renExpr( ("s" |-> int(2), "this" |-> ob("myglobal")), ("f" |-> int(3), ".label" |-> label(ob("mymethod"), 1) ), "+" ( "s" :: "f" ) ) .
rew renExpr( ("s" |-> int(2)), ("f" |-> int(3), ".label" |-> label(ob("mymethod"), 1) ), "+" ( "s" :: "f" ) ) .
rew renExpr( ("s" |-> int(2), "this" |-> ob("myglobal")), ("f" |-> int(3)), "+" ( "s" :: "f" ) ) .

rew getTrans(assign("x" ; "+" ("x" :: "d" ) ), "x" |-> int(4), "d" |-> int(3) ) .
rew appendTrans( getTrans(assign("x" ; "+" ("x" :: int(1) ) ) ) , getTrans(assign("z" ; "x") )  ) .
rew appendTrans( ( "x" |> "+"("x" :: int(1))), ("z" |> "x", "x" |> int(3) ) ) .


quit
