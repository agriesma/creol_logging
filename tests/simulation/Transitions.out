==========================================
rewrite in CREOL-SIMULATOR : getThis(("s" |-> int(2), "this" |-> ob(
    "myglobal"))) .
result String: "myglobal"
==========================================
rewrite in CREOL-SIMULATOR : getLabel(("s" |-> int(2), "this" |-> ob(
    "myglobal"))) .
result String: "nolabel"
==========================================
rewrite in CREOL-SIMULATOR : replacementMap(("s" |-> int(2), "this" |-> ob(
    "myglobal")), (".label" |-> label(ob("mymethod"), 1), "f" |-> int(3))) .
result TSubst: 
  ".label" |> "mymethod-1..label", 
  "f" |> "mymethod-1.f", 
  "s" |> "myglobal.s", 
  "this" |> "myglobal.this"
==========================================
rewrite in CREOL-SIMULATOR : replacementMap("s" |-> int(2), (".label" |->
    label(ob("mymethod"), 1), "f" |-> int(3))) .
result TSubst: 
  ".label" |> "mymethod-1..label", 
  "f" |> "mymethod-1.f", 
  "s" |> "global.s"
==========================================
rewrite in CREOL-SIMULATOR : replacementMap(("s" |-> int(2), "this" |-> ob(
    "myglobal")), "f" |-> int(3)) .
result TSubst: 
  "f" |> "nolabel.f", 
  "s" |> "myglobal.s", 
  "this" |> "myglobal.this"
==========================================
rewrite in CREOL-SIMULATOR : replacementMap(("s" |-> int(2), "this" |-> ob(
    "myglobal")), ("s" |-> int(2), ".label" |-> label(ob("mymethod"), 1), "f"
    |-> int(3))) .
result TSubst: 
  ".label" |> "mymethod-1..label", 
  "f" |> "mymethod-1.f", 
  "s" |> "mymethod-1.s", 
  "this" |> "myglobal.this"
==========================================
rewrite in CREOL-SIMULATOR : replace("&&"("<"("m" :: "mmax") :: "<"("-"("mfree"
    :: "t") :: "/"("m" :: "mrate"))), "m", "nte") .
result Expr: "&&"("<"("nte" :: "mmax") :: "<"("-"("mfree" :: "t") :: "/"("nte"
    :: "mrate")))
==========================================
rewrite in CREOL-SIMULATOR : renExpr(("s" |-> int(2), "this" |-> ob(
    "myglobal")), (".label" |-> label(ob("mymethod"), 1), "f" |-> int(3)), "+"(
    "s" :: "f")) .
result Expr: "+"("myglobal.s" :: "mymethod-1.f")
==========================================
rewrite in CREOL-SIMULATOR : renExpr("s" |-> int(2), (".label" |-> label(ob(
    "mymethod"), 1), "f" |-> int(3)), "+"("s" :: "f")) .
result Expr: "+"("global.s" :: "mymethod-1.f")
==========================================
rewrite in CREOL-SIMULATOR : renExpr(("s" |-> int(2), "this" |-> ob(
    "myglobal")), "f" |-> int(3), "+"("s" :: "f")) .
result Expr: "+"("myglobal.s" :: "nolabel.f")
==========================================
rewrite in CREOL-SIMULATOR : getTrans(assign("x" ; "+"("x" :: "d")), "x" |->
    int(4), "d" |-> int(3)) .
result TBinding: 
  "global.x" |> "+"("global.x" :: "nolabel.d")
==========================================
rewrite in CREOL-SIMULATOR : appendTrans(getTrans(assign("x" ; "+"("x" :: int(
    1)))), getTrans(assign("z" ; "x"))) .
result TSubst: 
  "x" |> "+"("x" :: int(1)), 
  "z" |> "+"("x" :: int(1))
==========================================
rewrite in CREOL-SIMULATOR : appendTrans(
  "x" |> "+"("x" :: int(1)), (
  "x" |> int(3), 
  "z" |> "x")) .
result TSubst: 
  "x" |> int(3), 
  "z" |> "+"("x" :: int(1))
Bye.
